src/box/lua/net_box.lua

```diff
@@ -698,9 +698,22 @@ end
 function remote_methods:_request(method, opts, format, stream_id, ...)
     local transport = self._transport
     local on_push, on_push_ctx, buffer, skip_header, return_raw, deadline
+
+    if opts ~= nil then
+        box.utils.check_param_table(opts, {buffer      = "cdata",
```

Comment:
I think that we should also ensure that the buffer has the right ffi type, see `checkibuf`, not just any `cdata`.